### PR TITLE
Implement Fmls_Se & Smov Opcodes

### PR DESCRIPTION
**Pull request by greggameplayer**

### CODE REVIEW

File: ChocolArm64/Instruction/AInstEmitSimdArithmetic.cs [hunk 1]
```diff
@@ -519,6 +524,15 @@ public static void Fmla_Ve(AILEmitterCtx Context)
                 Context.Emit(OpCodes.Add);
             });
         }
+        
+        public static void Fmls_Se(AILEmitterCtx Context)
+        {
+            EmitScalarTernaryOpByElemF(Context, () =>
+            {
+                Context.Emit(OpCodes.Mul);
+                Context.Emit(OpCodes.Add);
```
Comment (LDj3SNuD) [inline ChocolArm64/Instruction/AInstEmitSimdArithmetic.cs]: Context.Emit(OpCodes.Sub);

File: ChocolArm64/AOpCodeTable.cs [hunk 2]
```diff
@@ -371,6 +373,7 @@ static AOpCodeTable()
             SetA64("0x001110<<1xxxxx011011xxxxxxxxxx", AInstEmit.Smin_V,        typeof(AOpCodeSimdReg));
             SetA64("0x001110<<1xxxxx101011xxxxxxxxxx", AInstEmit.Sminp_V,       typeof(AOpCodeSimdReg));
             SetA64("0x001110<<1xxxxx100000xxxxxxxxxx", AInstEmit.Smlal_V,       typeof(AOpCodeSimdReg));
+            SetA64("0x001110000xxxxx001011xxxxxxxxxx", AInstEmit.Smov,          typeof(AOpCodeSimdIns));
```
Comment (LDj3SNuD) [inline ChocolArm64/AOpCodeTable.cs]: AInstEmit.Smov_S

File: ChocolArm64/Instruction/AInstEmitSimdMove.cs [hunk 3]
```diff
@@ -220,6 +220,15 @@ public static void Mvni_V(AILEmitterCtx Context)
         {
             EmitVectorImmUnaryOp(Context, () => Context.Emit(OpCodes.Not));
         }
+        
+        public static void Smov(AILEmitterCtx Context)
```
Comment (LDj3SNuD) [inline ChocolArm64/Instruction/AInstEmitSimdMove.cs]: Smov_S(AILEmitterCtx Context)

File: ChocolArm64/Instruction/AInstEmitSimdArithmetic.cs (hunk 1 above)
Comment (greggameplayer) [reply ChocolArm64/Instruction/AInstEmitSimdArithmetic.cs]: Instead of which context.emit ? both of them ?

Comment (LDj3SNuD) [reply ChocolArm64/Instruction/AInstEmitSimdArithmetic.cs]: The selected line ... (L#533).

File: ChocolArm64/AOpCodeTable.cs [hunk 4]
```diff
@@ -371,6 +373,7 @@ static AOpCodeTable()
             SetA64("0x001110<<1xxxxx011011xxxxxxxxxx", AInstEmit.Smin_V,        typeof(AOpCodeSimdReg));
             SetA64("0x001110<<1xxxxx101011xxxxxxxxxx", AInstEmit.Sminp_V,       typeof(AOpCodeSimdReg));
             SetA64("0x001110<<1xxxxx100000xxxxxxxxxx", AInstEmit.Smlal_V,       typeof(AOpCodeSimdReg));
+            SetA64("0x001110000xxxxx001011xxxxxxxxxx", AInstEmit.Smov_S,          typeof(AOpCodeSimdIns));
```
Comment (LDj3SNuD) [inline ChocolArm64/AOpCodeTable.cs]: Align typeof(AOpCodeSimdIns));

**Comment by AcK77** [comment]
@LDj3SNuD does it's look good to you now ?

**Comment by AcK77** [comment]
@MerryMage, @MS-DOS1999 and @gdkchan it's your turn I guess :)

**Comment by gdkchan** [comment]
The scalar implementations of FMINNM/FMAXNM are doing basically the same thing (ignoring NaNs), I think that those instructions should be fixed and FMINNM_V should be implemented the right way by doing the NaN checks like a real ARM cpu. The min/max instructions are already using a method to handle the -0 cases anyway, so we should just create another one that also handles  NaNs.

**Comment by gdkchan** [comment]
#296 implements `Fminnm_V`, please remove it from this PR.

**Comment by greggameplayer** [comment]
@gdkchan done

**Comment by emmauss** [comment]
Any changes to be done here?

**Comment by gdkchan** [comment]
Closing in favor of #449.

**Pull request closed without merging**